readthedocs/projects/models.py

```diff
@@ -677,6 +723,35 @@ def custom_path_prefix(self):
             return self.urlconf.split("$", 1)[0]
         return None
 
+    @cached_property
+    def subproject_prefix(self):
+        """
+        Returns the path prefix of a subproject.
+
+        This normally is ``/projects/<subproject-alias>/``,
+        but if the project has a custom URL pattern,
+        that pattern will be used.
+
+        Returns `None` if the project isn't a subproject.
+        """
+        parent_relationship = self.parent_relationship
+        if not parent_relationship:
+            return None
+
+        parent_project = parent_relationship.parent
+        if parent_project.urlpattern_subproject:
+            subproject_prefix = urlpattern_to_plain_text(
+                parent_project.urlpattern_subproject
+            )
+            # The filename isn't part of the subproject prefix,
+            # so we remove it from the pattern.
+            index = subproject_prefix.find("{filename}")
+            subproject_prefix = subproject_prefix[:index]
```

Comment:
I don't fully understand why this is required? Should we make this function return the value with the filename to keep it simpler?

Reply:
This method  returns the subproject prefix, the filename isn't part of the prefix.